script/PSQL.py

```diff
@@ -0,0 +1,25 @@
+import psycopg2
+import datetime
+
+# user = "postgres"
+# password = "sL6IpcVZKv57HcZZTX3x"
+# host = "package-picker-test.c7etfrntf9yq.us-east-1.rds.amazonaws.com"
+
+conn_string = "host='localhost' dbname='postgres' user='postgres' password='secret'"
+def connectToDB():
+    # Connect to the database
+    db = psycopg2.connect(conn_string)
+    return db
+
+def insertToApplication(db, url, followers, appName):
+    # Get a cursor for executing queries
+    cur = db.cursor()
+    # Upsert a row into the applications table
+    cur.execute("INSERT INTO applications (url, name, followers, retrieved) VALUES (%s, %s, %s, %s) ON CONFLICT ON CONSTRAINT unique_url_and_hash DO UPDATE SET (url, name, followers, retrieved) = (EXCLUDED.url, EXCLUDED.name, EXCLUDED.followers, EXCLUDED.retrieved)", (url, appName, followers, datetime.datetime.now()))
```

Comment:
Need to set `hash` to a non-null value, otherwise the uniqueness constraint will never be violated and we'll end up with duplicate rows.

Reply:
I just added a NULL constraint to the tables so you'll no longer be able to insert rows with missing required values (see https://sourcecoderecommender.atlassian.net/wiki/spaces/SCR/pages/27164801/Database+schema for more details)